plugin-server/src/kafka/producer.ts

```diff
@@ -0,0 +1,105 @@
+import {
+    ClientMetrics,
+    HighLevelProducer as RdKafkaProducer,
+    NumberNullUndefined,
+    ProducerGlobalConfig,
+} from 'node-rdkafka'
+
+import { status } from '../utils/status'
+
+// Kafka production related functions using node-rdkafka.
+export const createKafkaProducer = async (config: ProducerGlobalConfig) => {
+    const producer = new RdKafkaProducer({
+        // milliseconds to wait before sending a batch. The default is 0, which
```

Comment:
question: while I initially thought we would create one batch per 20ms, my understanding of `linger.ms` is that each new message resets that timeout, we only trigger a batch if no new message has been submitted to the producer in 20ms.

Value looks fine (as we'll manually flush anyway, and it runs fine on prod), just wanted to confirm my understanding.